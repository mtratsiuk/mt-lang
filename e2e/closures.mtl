(def (createPoint x y)
  (def (point prop)
    (cond
      ((= prop "x") x)
      ((= prop "y") y))))

(def point (createPoint 10 20))

(print (point "x"))
(print (point "y"))
