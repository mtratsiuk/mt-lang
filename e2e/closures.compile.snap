const __mtl_curry = f => (...args) => {
  if (args.length >= f.length) {
    return f(...args);
  }

  return __mtl_curry(f.bind(null, ...args));
};

const createPoint = __mtl_curry((x, y) => {
  return ((prop) => {
    return (() => {
      if ((prop === "x")) {
        return x;
      } else if ((prop === "y")) {
        return y;
      }
    })();
  });
});

const point = createPoint(10, 20);

console.log(String(point("x")));

console.log(String(point("y")));
